yaml/private/internal.nim

```diff
@@ -5,16 +5,24 @@
 #    distribution, for details about the copyright.
 
 template internalError*(s: string) =
+  # Note: to get the internal stacktrace that caused the error
+  # compile with the `d:debug` flag.
   when not defined(release):
     let ii = instantiationInfo()
     echo "[NimYAML] Error in file ", ii.filename, " at line ", ii.line, ":"
     echo s
     when not defined(JS):
       echo "[NimYAML] Stacktrace:"
-      try: writeStackTrace()
+      try:
+        writeStackTrace()
+        when defined(debug):
```

Comment:
I think this is not necessary given that we are already in `when not defined(release)`. Also, shouldn't you check whether `exc.parent` is `nil`?

Reply:
Ugh, didn't event think of checking release... Good point, also regarding checking for nil!